Source/WebCore/rendering/RenderBox.cpp

```diff
@@ -1396,13 +1396,28 @@ void RenderBox::clearOverridingLogicalWidthLength()
 
 FlowRelativeDirection RenderBox::physicalToFlowRelativeDirectionMapping(PhysicalDirection direction) const
 {
-    auto isHorizontalWritingMode = style().isHorizontalWritingMode();
-    auto isLeftToRightDirection = style().isLeftToRightDirection();
+    auto determineFormattingContextStyle = [&]() -> const RenderStyle& {
+        if (isFlexItem())
+            return parent()->style();
+        ASSERT_NOT_IMPLEMENTED_YET();
+        return style();
+    };
```

Comment:
formatting contexts do not have styles (formatting context roots do), FCs have directionalities (inline, block) among other things, but you can't say e.g. this FC has the background color of green.

Reply:
I changed the naming here to reference the FC root instead since that is the intention of the helper. I believe this should be the correct style to use when determining the `isFlippedBlocksWritingMode` and `isLeftToRightDirection` below, but let me know if you think that's not right